libs/core/src/main/java/org/opensearch/core/rest/RestStatus.java

```diff
@@ -559,4 +567,30 @@ public static RestStatus status(int successfulShards, int totalShards, ShardOper
     public static RestStatus fromCode(int code) {
         return CODE_TO_STATUS.get(code);
     }
+
+    public static boolean isValidRestCode(final int code) {
+        return fromCode(code) != null;
+    }
```

Comment:
Does this need to be public?

Reply:
It's introduced as an auxiliary method to check for valid rest code. Just extending the `RestStatus` enum here. In earlier iteration, I used this to validate whether the settings provided by user consists of only valid rest codes.

Should it stay? I don't see any need now since we dropped the settings idea. But good to have as a part of API.